core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutineCompiler.java

```diff
@@ -334,8 +335,8 @@ public BytecodeNode visitBlock(IrBlock node, Scope scope)
             LabelNode breakLabel = new LabelNode("break");
 
             if (node.label().isPresent()) {
-                continueLabels.put(node.label().get(), continueLabel);
-                breakLabels.put(node.label().get(), breakLabel);
+                checkState(continueLabels.putIfAbsent(node.label().get(), continueLabel) == null, "continue label for loop label %s already exists", node.label().get());
```

Comment:
Use `verify()` for these, like the other checks. This condition does not match the description for `IllegalStateException`.

Reply:
fixed!